src/ssh.ext.c

```diff
@@ -81,22 +86,57 @@ int show_remote_processes(ssh_session session)
   return SSH_OK;
 }
 
+$R sshQ_ChannelD__initG_local (sshQ_Channel self, $Cont c$cont) {
+    int err = 0;
+    ssh_channel channel = ssh_channel_new(self->_ssh_session);
+    if (channel == NULL) {
+        LOG_ERR("Failed to create SSH channel");
+        printf("ssh_get_error: %s\n\n" , ssh_get_error(self->_ssh_session));
+        return $R_CONT(c$cont, B_None);
+    }
+
+    printf("\t%s channel: %p\n", __FUNCTION__, channel);
+    self->_ssh_channel = channel;
+    printf("\t%s self->_ssh_channel:\t%p\n", __FUNCTION__, self->_ssh_channel);
+
+    err = ssh_channel_open_session(channel);
+    if (err != SSH_OK) {
+        printf("\t%s ssh_channel_open_session() error (%d)\n", __FUNCTION__, err);
+        return $R_CONT(c$cont, B_None);
+    }
+    if (ssh_channel_request_exec(channel, "touch /tmp/bla")) {
+        printf("\t%s Error executing '%s' : %s\n", __FUNCTION__, "touch /tmp/bla", ssh_get_error(self->_ssh_session));
+        // free(buffer);
+        // ssh_channel_free(channel);
+        return $R_CONT(c$cont, B_None);
+    }
+    // free(buffer);
+    // select_loop(session, channel);
+    // ssh_channel_free(channel);
+
+    return $R_CONT(c$cont, B_None);
+}
+
 $R sshQ_ClientD__initG_local (sshQ_Client self, $Cont c$cont) {
     ssh_session session = ssh_new();
     if (session == NULL) {
-        //log_error("Failed to create SSH session");
+        LOG_ERR("Failed to create SSH session");
         return $R_CONT(c$cont, B_None);
     }
-    printf("session: %p\n", session);
-    self->_ssh_session = toB_u64((unsigned long)session);
-    printf("init self->session: %p\n", self->_ssh_session);
+    // casting via toB_u64((unsigned long)..) leads to an invalid value in self->_ssh_session
+    // self->_ssh_session = toB_u64((unsigned long)session);
+    // instead do direct assignment
+    self->_ssh_session = session;
```

Comment:
@plajjan this one took me a bit to realize. Doing `toB_u64((unsigned long)session)` makes `self->_ssh_session` have a different value compared to the `session` pointer

Reply:
tldr: `toB_u64()` saves into block. Needed to be done if the variable/value has to be accessed from acton instead of only C

Reply:
Yes exactly, toB_u64() takes the unsigned long value and creates a B_u64 type, which is the Acton boxed type for a u64, like this (from https://github.com/actonlang/acton/blob/main/base/builtin/u64.h):
```c
struct B_u64 {
    struct B_u64G_class *$class;
    unsigned long val;
};
```

you can use `toB_u64()` - it's just that you then have to use `fromB_u64` when reading it. You can see in https://github.com/actonlang/acton/blob/main/base/builtin/u64.c#L71C1-L73C2 that it just read the val:
```c
unsigned long fromB_u64(B_u64 w) {
    return w->val;
}
```

as long as you are consistent, either way should be fine. Like you said, if we want to be able to read the value from Acton code in a comprehensible manner, then it must be an actual Acton B_u64 type.

This might very well change in the future too as we are adding better support for unboxed values, but more on that some other time :)